screed/tests/test_attriberror.py

```diff
@@ -0,0 +1,85 @@
+import screed
+from screed.DBConstants import fileExtension
+import os
+
+
+class nostring:
+    def __init__(self):
+        self.exists = True
+
+    def __str__(self):
+        raise AttributeError
```

Comment:
As far as I can see it is impossible to delete `__str__` from both new style and old style objects. Therefor the try/except should be deleted and any code that was intended to test it.

``` python
In [2]: class test(object):                                                                                                                 
   ...:     
   ...: 
   ...:     pass
   ...: 

In [3]: 

In [3]: dir(test)
Out[3]: 
['__class__',
 '__delattr__',
 '__dict__',
 '__doc__',
 '__format__',
 '__getattribute__',
 '__hash__',
 '__init__',
 '__module__',
 '__new__',
 '__reduce__',
 '__reduce_ex__',
 '__repr__',
 '__setattr__',
 '__sizeof__',
 '__str__',
 '__subclasshook__',
 '__weakref__']

In [13]: instance = test()                                                                                                                  

In [14]: str(instance)                                                                                                                      
Out[14]: '<__main__.test object at 0x7ffe350f7810>'

In [15]: del test.__str__
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-15-1e4d1055a40b> in <module>()
----> 1 del test.__str__

AttributeError: __str__

In [16]: class test2:                                                                                                                       
   
    pass
   ....: 

In [17]: instance2 = test2()                                                                                                                

In [18]: dir(instance2)
Out[18]: ['__doc__', '__module__']

In [19]: str(instance2)
Out[19]: '<__main__.test2 instance at 0x7ffe35101560>'

In [20]: del instance2.__str__
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-20-7bec2f1814ad> in <module>()
----> 1 del instance2.__str__

AttributeError: test2 instance has no attribute '__str__'

```